### ISSUE TITLE
Add Travis CI support

### ISSUE THREAD

**Issue by elast0ny** [elast0ny/mem_file#8]
As suggested by @llogiq , mem_file would probably benefit from the features provided by https://travis-ci.org/

I will have to set up compilation on the supported targets at a minimum and maybe try to see how testing could be implemented.

Testing might prove to be a bit tricky since mem_file is a library aimed for multi-process environments and it sounds like a headache to do automated testing for that. Issues that come to mind:

- Granted that you can launch multiple separate processes in a single cargo test, how do you validate:
    -Memory is actually shared
    -Locking mechanism work properly (the two process cant both hold a mutex for example)
    -Things are getting cleaned up/closed properly after a non-crashing/non-frozen run

- In case of error for any of these testcases, your testing framework might have to deal with crashes processes, frozen processes waiting forever on locks, etc.. on a per-OS basis.

**Comment by elast0ny** [comment]
Added basic travis config https://github.com/elast0ny/mem_file/commit/6746e3193b0de0e48ee1930f14d49b6a28d784a5

Still need to figure out how to get this to compile for Windows (if possible) and then think about adding Mac support.

**Comment by llogiq** [comment]
Travis doesn't have Windows support AFAIK. I personally use AppVeyor for that.

Also your travis script doesn't *need* to only run `cargo test`, you could also `cargo run --example sender & cargo run --example receiver` and make `sender` and `receiver` share some memory. Verify by having `sender` write some data into shared memory and `receiver` check that the right thing has been written.

**Comment by elast0ny** [comment]
@llogiq : Maybe my last comment wasnt clear enough but essentially what i'm wondering is, regardless of if you run cargo test or run --example X, does travis allow running those commands in parallel ? If it does, is there also a way to tell travis that these two commands depend on each other ? 

(Ie: if my sender spawns and wait forever for the receiver which has crashed, travis should kill the sender and return a failure)

From what I can tell, thats not how travis works and all commands in the "script" field *might* run in parallel (for performance reasons) and shouldnt depend on each other.

If there is somehow a way in cargo to say, run these two files in seperate processes and if any fail, consider this test a fail, then we could use that cargo feature in travis as a single test with a single cargo command.

As a side note, i now have windows and linux x86/64 builds working on travis and can just un-comment the lines for OSX when a Mac implementation will be merged into the project

**Comment by elast0ny** [comment]
Travis-ci is now set up for builds on : 
- Linux
-MacOS
-Windows gnu

If Windows MSVC builds are required, I will have to look into Appveyor.
